book/src/g_object_memory_management.md

```diff
@@ -195,5 +195,4 @@ When we set `gtk_box` as child of `window`, `window` keeps a strong reference to
 Until we close the `window` it keeps `gtk_box` and with it the buttons alive.
 Since our application has only one window, closing it also means exiting the application.
 
-As long as you use weak references whenever possible you will find it perfectly doable to avoid memory cycles within your application.
-If that is ensured, you can rely on GTK to properly manage the memory of GObjects you pass to it.
+As long as you use weak references whenever possible, you will find it perfectly doable to avoid memory cycles within your application, and you can rely on GTK to properly manage the memory of GObjects you pass to it.
```

Comment:
```suggestion
As long as you use weak references whenever possible, you will find it perfectly doable to avoid memory cycles within your application.
Without memory cycles, you can rely on GTK to properly manage the memory of GObjects you pass to it.
```

How about this?

Reply:
Looks great! :D